### PR TITLE
Optimize output from RGDR

### LINKED ISSUE #86: When/how should RGDR fail if a lag has no significant clusters
In PR #85 support for analyzing multiple precursor lags w/ RGDR was added. When a lag does not contain any clusters, RGDR will raise an error. However, a discussion was started on _when_ to raise this error. Either immediately once a certain lag was analyzed and did not contain a cluster, or once all lags have been analyzed and one or more lags do not contain a cluster.

There are pro's and con's to either.

Why raise an error **immediately**:

- if the clustering takes long, and it fails at the first lag, then users won't have to wait for it to complete to be notified of the error. 

Why raise an error **after processing all lags**:

- Users do not have to iteratively remove lags without clusters.
- Users can find "predictability windows", e.g. there is only significant correlation for a 1 month window half a year ahead of the target time.

In the current implementation, RGDR will analyze all lags first, and only then raise an error.

### LINKED ISSUE #92: Reshape the RGDR returned values to be [samples, features]
Currently the RGDR module will return a `dataarray` with dimensions `[cluster_labels, anchor_year]` after calling `transform` (e.g. `rgdr.transform(precursor_field)`). For most of the popular machine learning packages, e.g. `scikit-learn`, the output from the dimensionality reduction method is always in the shape `[samples, features]` (e.g. [PCA in sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) and the models also need input to be organized in this way (e.g. [GraidentBoostingMachine in sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html#sklearn.ensemble.GradientBoostingRegressor)).

It is nice to have the output from RGDR to be return with the shape `[anchor_year, cluster_labels]`, which is compatible with sklearn models.

### PR DESCRIPTION
**Pull request by geek-yang**
Tiny updates to RGDR module:

- Return `dataarray` with the shape `["anchor_year", "cluster_labels"]` to be compatible with sklearn models (require inputs in the format of [samples, features])
- Add attributes to the returned values (e.g. latitude/longitude refer to the geographical center of each cluster)
- Replace errors with warnings when there is only one cluster found

This PR closes #92 and closes #86

### CODE REVIEW

**Comment by semvijverberg** [comment]
Dear @geek-yang,

Small question. If you make a pull request, it means you are asking for a review by a team member right? 

I noticed the first two points have been addressed nicely. 
- Return dataarray with the shape ["anchor_year", "cluster_labels"] to be compatible with sklearn models (require inputs in the format of [samples, features])
- Add attributes to the returned values (e.g. latitude/longitude refer to the geographical center of each cluster)

But I did not see any changes in 'files changed', that tackle the last point:
- Replace errors with warnings when there is only one cluster found. 

Cheers,
Sem

**Comment by geek-yang** [comment]
> Dear @geek-yang,
> 
> Small question. If you make a pull request, it means you are asking for a review by a team member right?
> 
> I noticed the first two points have been addressed nicely.
> 
> * Return dataarray with the shape ["anchor_year", "cluster_labels"] to be compatible with sklearn models (require inputs in the format of [samples, features])
> * Add attributes to the returned values (e.g. latitude/longitude refer to the geographical center of each cluster)
> 
> But I did not see any changes in 'files changed', that tackle the last point:
> 
> * Replace errors with warnings when there is only one cluster found.
> 
> Cheers, Sem
> 
> -- Ah I know see this PR is 'still a work in progress' and it has not been set to 'Ready for review'!
> 
> Learned something again today! :)

Haha, thanks for your early review @semvijverberg . Indeed it is still a working progress. When it is ready for review, I will click the button ready for review and add you as the reviewer (see the right corner). Then you can check the code, leave your comment and let me know if you give me a pass or not. It will be done soon. But thanks for the early notice 😆 .

**Comment by semvijverberg** [comment]
Hey @geek-yang,

Small remark. The suggestion to remove the intervals is not really needed. In a larger pipeline there is nothing wrong with not finding cluster at a specific lag. 

Hence, my suggestion would be to remove the sentence "Please remove these intervals from the model before continuing." at line 118. 

It is my opinion that after this minor change the PR can be merged. 

Thanks for the work:)

**Comment by geek-yang** [comment]
> 

Thanks for your review @semvijverberg  and thank you for spotting this. It is deleted now. If everything looks good for now, could you please give me a pass via the "review changes" process (click "Files changed" and click "review changes", then you can approve this PR)?

**Review by semvijverberg — APPROVED** [review]

File: s2spy/rgdr/rgdr.py [hunk 1]
```diff
@@ -534,12 +534,16 @@ def transform(self, data: xr.DataArray) -> xr.DataArray:
 
         # Add the geographical centers for later alignment between, e.g., splits
         reduced_data = utils.geographical_cluster_center(data, reduced_data)
+        # Include explanations about geographical centers as attributes
+        reduced_data.attrs['data'] = "Clustered data with Response Guided Dimensionality Reduction."
+        reduced_data.attrs['coordinates'] = "Latitudes and longitudes are geographical centers associated with clusters."
 
         # Remove the '0' cluster
         reduced_data = reduced_data.where(reduced_data["cluster_labels"] != "0").dropna(
             dim="cluster_labels"
```
Comment (semvijverberg) [inline s2spy/rgdr/rgdr.py:543]: Here I'm adding a comment for testing.

File: s2spy/rgdr/rgdr.py [hunk 2]
```diff
@@ -534,12 +534,16 @@ def transform(self, data: xr.DataArray) -> xr.DataArray:
 
         # Add the geographical centers for later alignment between, e.g., splits
         reduced_data = utils.geographical_cluster_center(data, reduced_data)
+        # Include explanations about geographical centers as attributes
+        reduced_data.attrs['data'] = "Clustered data with Response Guided Dimensionality Reduction."
+        reduced_data.attrs['coordinates'] = "Latitudes and longitudes are geographical centers associated with clusters."
 
         # Remove the '0' cluster
         reduced_data = reduced_data.where(reduced_data["cluster_labels"] != "0").dropna(
```
Comment (semvijverberg) [inline s2spy/rgdr/rgdr.py:542]:
```suggestion
        reduced_data = reduced_data.where(reduced_data["cluster_labels"] != "0")
```

**Pull request merged**